src/catalog/src/memory/objects.rs

```diff
@@ -821,6 +823,31 @@ impl CatalogItem {
         }
     }
 
+    /// Collects the identifiers of the objects used by this [`CatalogItem`].
+    ///
+    /// Like [`CatalogItem::references()`] but also includes objects that are not directly
+    /// referenced. For example this will include any catalog objects used to implement functions
+    /// and casts in the item.
+    pub fn uses(&self) -> BTreeSet<GlobalId> {
+        let mut uses = self.references().0.clone();
```

Comment:
I was hoping to avoid this clone in the final version but I never thought far enough about how to do that. It's probably not a huge deal, but I just wanted to call it out.